src/Build/BackEnd/Components/SdkResolution/SdkResolverService.cs

```diff
@@ -48,12 +51,17 @@ internal class SdkResolverService : ISdkResolverService
         /// <summary>
         /// Stores the list of manifests of specific SDK resolvers which could be loaded.
         /// </summary>
-        private IList<SdkResolverManifest> _specificResolversManifestsRegistry;
+        private IReadOnlyList<SdkResolverManifest> _specificResolversManifestsRegistry;
 
         /// <summary>
         /// Stores the list of manifests of general SDK resolvers which could be loaded.
         /// </summary>
-        private IList<SdkResolverManifest> _generalResolversManifestsRegistry;
+        private IReadOnlyList<SdkResolverManifest> _generalResolversManifestsRegistry;
+
+#if DEBUG
```

Comment:
Can we use "Extended Unit Under Test" pattern instead of conditional code within production code?